File: Julia/conversor.jl
function decimal_para_binario(decimal::Int)
    decimal == 0 && return "0"

    binario = ""
    numero = decimal
    while numero > 0
        binario = string(numero % 2) * binario
        numero ÷= 2
    end
    return binario
end

println("Digite um numero decimal: ")
decimal = parse(Int, readline())
println("Binario: ", decimal_para_binario(decimal))

File: Rust/conversor.rs
use std::io;

fn decimal_para_binario(decimal: u32) -> String {
    if decimal == 0 {
        return String::from("0");
    }

    let mut numero = decimal;
    let mut binario = String::new();
    while numero > 0 {
        binario.insert(0, char::from_digit(numero % 2, 10).unwrap());
        numero /= 2;
    }
    binario
}

fn main() {
    println!("Digite um numero decimal: ");
    let mut entrada = String::new();
    io::stdin().read_line(&mut entrada).unwrap();
    let decimal: u32 = entrada.trim().parse().unwrap();

    println!("Binario: {}", decimal_para_binario(decimal));
}
